library/alloc/src/str.rs

```diff
@@ -615,50 +603,86 @@ pub unsafe fn from_boxed_utf8_unchecked(v: Box<[u8]>) -> Box<str> {
     unsafe { Box::from_raw(Box::into_raw(v) as *mut str) }
 }
 
-/// Converts the bytes while the bytes are still ascii.
+/// Converts leading ascii bytes in `s` by calling the `convert` function.
```

Comment:
I wonder what the tradeoff between only processing a prefix and mode-switching between ascii and unicode chunks. The utf8 validation loop does the latter.

But that's doesn't have to happen in this PR.

Reply:
Worth investigating, could be beneficial for some european languages with the occasional umlaut among mostly ascii texts.